bin/oneoff/back_populate_user_proficiencies

```diff
@@ -0,0 +1,210 @@
+  proficiency.functions_d3_count += line[41].to_i
+  proficiency.functions_d4_count += line[42].to_i
+  proficiency.functions_d5_count += line[43].to_i
+
+  proficiency.functions_with_params_d1_count += line[44].to_i
+  proficiency.functions_with_params_d2_count += line[45].to_i
+  proficiency.functions_with_params_d3_count += line[46].to_i
+  proficiency.functions_with_params_d4_count += line[47].to_i
+  proficiency.functions_with_params_d5_count += line[48].to_i
+
+  proficiency.conditionals_d1_count += line[49].to_i
+  proficiency.conditionals_d2_count += line[50].to_i
+  proficiency.conditionals_d3_count += line[51].to_i
+  proficiency.conditionals_d4_count += line[52].to_i
+  proficiency.conditionals_d5_count += line[53].to_i
+
+  proficiency
+end
+
+# @param proficiency [UserProficiency] a UserProficiency populated with
+#   historical timestamps and the sum of historical and incremental level counts
+# @param line [Array] an array giving historical UserProficiency values
+# @return [DateTime | nil] the best available datetime at which basic
+#   proficiency was obtained (nil if not yet obtained)
+def compute_basic_proficiency_at(proficiency, line)
+  # Determine whether the historical behavior (specified in line) sufficed to
+  # obtain basic proficiency.
+  proficiency_from_history = increment_user_proficiency(
+    UserProficiency.new, line
+  ).basic_proficiency?
+  # If it did, return TIME_CUTOFF.
+  return TIME_CUTOFF if proficiency_from_history
+  # Otherwise, if the incremental behavior sufficed by itself - as observed
+  # through proficiency.basic_proficiency_at - return the existing timestamp.
+  return proficiency.basic_proficiency_at if proficiency.basic_proficiency_at
+  # Otherwise, if the historical and incremental behavior together suffice,
+  # return last_progress_at.
+  return proficiency.last_progress_at if proficiency.basic_proficiency?
+  # Otherwise, basic proficiency has not been obtained, so return nil.
+  return nil
+end
+
+CSV.foreach('/tmp/user_proficiency.csv').with_index do |line, line_num|
+  puts "PROCESSING: line #{line_num}..." if line_num % 1000 == 0
+
+  begin
+    user_id = line[0].to_i
+
+    proficiency = UserProficiency.find_or_create_by(user_id: user_id)
+
+    # The level counts are the sum of historical level counts (given by line)
+    # and incremental level counts (given by proficiency).
+    proficiency = increment_user_proficiency(proficiency, line)
+    # We set basic_proficiency_at before the other timestamps, as it can depend
+    # on historical timestamp values.
+    proficiency.basic_proficiency_at = compute_basic_proficiency_at(
+      proficiency, line)
+    proficiency.created_at = TIME_NOW unless proficiency.created_at
+    proficiency.updated_at = TIME_NOW
+    # Though not accurate, TIME_CUTOFF is the most accurate estimate we have.
+    proficiency.last_progress_at = TIME_CUTOFF unless proficiency.last_progress_at
+
+    proficiency.save!
+  rescue Exception => e
```

Comment:
No idea. But I've learned not to expect our production data to behave.

Aside, the ensure is to make sure we have a checkpoint if the script is cancelled (e.g., by user `ctrl-c` or otherwise).